usr/src/lib/libnsl/rpc/svc_generic.c

```diff
@@ -106,6 +107,221 @@ __svc_free_xprtlist(void)
+			    "%lu: Failed open", nettype, versnum);
+			error(error_buf, sv_errno);
+			continue;
+		}
+
+		/* LINTED E_BAD_PTR_CAST_ALIGN */
+		bind_addr = (struct t_bind *)t_alloc(fd, T_BIND, T_ADDR);
+		if ((bind_addr == NULL)) {
+			sv_errno = errno;
+			(void) snprintf(error_buf,  256, "couldn't register %s "
+			    "%lu: Failed allocation", nettype, versnum);
+			error(error_buf, sv_errno);
+			(void) t_close(fd);
+			continue;
+		}
+		/* LINTED E_BAD_PTR_CAST_ALIGN */
+		reqb = (struct t_bind *)t_alloc(fd, T_BIND, T_ADDR);
+		if ((reqb == NULL)) {
+			sv_errno = errno;
+			(void) snprintf(error_buf,  256, "couldn't register %s "
+			    "%lu: Failed allocation", nettype, versnum);
+			error(error_buf, sv_errno);
+			(void) t_close(fd);
+			(void) t_free((char *)bind_addr, T_BIND);
+			continue;
+		}
+
+		if (strcmp(nconf->nc_protofmly, NC_INET6) == 0) {
+			struct sockaddr_in6 *sin;
+			reqb->qlen = 1;
+			reqb->addr.len = sizeof (struct sockaddr_in6);
+			/* LINTED E_BAD_PTR_CAST_ALIGN */
+			sin = (struct sockaddr_in6 *)reqb->addr.buf;
+			(void) memset((void *)sin, 0,
+			    sizeof (struct sockaddr_in6));
+			sin->sin6_family = AF_INET6;
+			sin->sin6_port = port;
+		} else if (strcmp(nconf->nc_protofmly, NC_INET) == 0) {
+			struct sockaddr_in *sin;
+			reqb->qlen = 1;
+			reqb->addr.len = sizeof (struct sockaddr_in);
+			/* LINTED E_BAD_PTR_CAST_ALIGN */
+			sin = (struct sockaddr_in *)reqb->addr.buf;
+			(void) memset((void *)sin, 0,
+			    sizeof (struct sockaddr_in));
+			sin->sin_family = AF_INET;
+			sin->sin_port = port;
+		} else if (strcmp(nconf->nc_protofmly, NC_LOOPBACK) == 0) {
+			(void) t_free((char *)bind_addr, T_BIND);
+			(void) t_free((char *)reqb, T_BIND);
+			bind_addr = NULL;
+			reqb = NULL;
+		} else {
+			(void) t_close(fd);
+			(void) t_free((char *)bind_addr, T_BIND);
+			(void) t_free((char *)reqb, T_BIND);
+			(void) syslog(LOG_ERR, "Unhandled protofmly %s, %s\n",
+				nconf->nc_protofmly, nettype);
+			continue;
+		}
+
+		if (t_bind(fd, reqb, bind_addr) == -1) {
+			error("Bind failed", errno);
+			(void) t_close(fd);
+			if (bind_addr != NULL) {
+				(void) t_free((char *)bind_addr, T_BIND);
+				(void) t_free((char *)reqb, T_BIND);
+			}
+			continue;
+		}
+
+		if ((xprt = svc_tli_create(fd, nconf, bind_addr, 0, 0)) ==
+		    NULL) {
+			error("Service creation failed", errno);
+			(void) t_unbind(fd);
+			(void) t_close(fd);
+			if (bind_addr != NULL) {
+				(void) t_free((char *)bind_addr, T_BIND);
+				(void) t_free((char *)reqb, T_BIND);
+			}
+			continue;
+		}
+		(void) rpcb_unset(prognum, versnum, nconf);
+		if (svc_reg(xprt, prognum, versnum, dispatch, nconf) == FALSE) {
+			error("Failed to register service", errno);
+			(void) t_unbind(fd);
+			(void) t_close(fd);
+			SVC_DESTROY(xprt);
+			xprt = NULL;
+		} else {
+			num_reg++;
+		}
+
+		if (xprt && !__svc_add_to_xlist(&_svc_xprtlist, xprt,
+		    &xprtlist_lock)) {
+			(void) syslog(LOG_ERR, "svc_create_port: no memory");
+			return (0);
```

Comment:
Don't we still need to call __rpc_endconf() here? Otherwise aren't we leaking the memory allocated for handle?